nx-X11/programs/Xserver/hw/nxagent/Args.c

```diff
@@ -1047,6 +1047,24 @@ int ddxProcessArgument(int argc, char *argv[], int i)
     return 0;
   }
 
+  /*
+   * Disable Xinerama (i.e. fake it in Screen.c) if somehow Xinerama support
+   * has been disabled on the cmdline.
+   */
+#if defined(PANORAMIX) && defined(RANDR)
+  if ((noPanoramiXExtension || PanoramiXExtensionDisabledHack) && noRRXineramaExtension)
+    nxagentChangeOption(Xinerama, 0);
+#else
+#ifdef RANDR
+  if (noRRXineramaExtension)
+    nxagentChangeOption(Xinerama, 0);
+#endif
+#ifdef PANORAMIX
+  if (noPanoramiXExtension || PanoramiXExtensionDisabledHack)
+    nxagentChangeOption(Xinerama, 0);
+#endif
+#endif
+
   return 0;
```

Comment:
Hi Uli,

On  Mi 15 Jun 2016 14:02:59 CEST, Ulrich Sibiller wrote:

>> +   * has been disabled on the cmdline.
>> +   */
>> +#if defined(PANORAMIX) && defined(RANDR)
>> +  if ((noPanoramiXExtension || PanoramiXExtensionDisabledHack) &&  
>> noRRXineramaExtension)
>> +    nxagentChangeOption(Xinerama, 0);
>> +#else
>> +#ifdef RANDR
>> +  if (noRRXineramaExtension)
>> +    nxagentChangeOption(Xinerama, 0);
>> +#endif
>> +#ifdef PANORAMIX
>> +  if (noPanoramiXExtension || PanoramiXExtensionDisabledHack)
>> +    nxagentChangeOption(Xinerama, 0);
>> +#endif
>> +#endif
>> +
>>    return 0;
>
> This looks complicated.  I think it can be solved easier. At least  
> the last ifdef kan be replaced by an else.

How is this, then?

```
   if (
#ifdef PANORAMIX
      (noPanoramiXExtension || PanoramiXExtensionDisabledHack) &&
#endif
#ifdef RANDR
      (noRRXineramaExtension) &&
#endif
      (nxagentOption(Xinerama) == 1) /* to avoid an empty condition */
      )
     nxagentChangeOption(Xinerama, 0);
```

Mike
-- 

DAS-NETZWERKTEAM
mike gabriel, herweg 7, 24357 fleckeby
mobile: +49 (1520) 1976 148
landline: +49 (4354) 8390 139

GnuPG Key ID 0x25771B31
mail: mike.gabriel@das-netzwerkteam.de, http://das-netzwerkteam.de

freeBusy:
https://mail.das-netzwerkteam.de/mailxchange/kronolith/fb.php?u=m.gabriel%40das-netzwerkteam.de